### PR TITLE
[WIP] Updates to attester service

### PR DESCRIPTION
**Pull request by spble**
## Issue Addressed

#255 

## Proposed Changes

Restructures attester service & block producer according to new duties model.

 - Correctly naming AttestationData structs
 - Including the full `Keypair` in the validator client duties, instead of just the `PublicKey`
 - Remove _most_ of the legacy polling functionality in the attester/block producer services (including slot clock & duties manager)
 - Started removal of `FreeAttestation`s (now just `Attestations`)

### CODE REVIEW

File: eth2/attester/src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,12 @@
 pub mod test_utils;
```
Comment (AgeManning) [inline eth2/attester/src/lib.rs]: I had planned on leaving the eth2 crate alone and building a miniature, `Attester` in the validator client. 

It seems you have got it compiling. @paulhauner - Is this going to mess with any of your other harness if `attester` and `BlockProducer` gets modified?

File: eth2/types/src/attestation_duty.rs [hunk 2]
```diff
@@ -6,4 +6,5 @@ pub struct AttestationDuty {
     pub slot: Slot,
     pub shard: Shard,
     pub committee_index: usize,
+    pub validator_index: usize,
```
Comment (AgeManning) [inline eth2/types/src/attestation_duty.rs]: Is this needed?

File: validator_client/src/duties/mod.rs [hunk 3]
```diff
@@ -40,8 +40,9 @@ pub enum Error {
 /// This keeps track of all validator keys and required voting slots.
 pub struct DutiesManager<U: BeaconNode> {
     pub duties_map: RwLock<EpochDutiesMap>,
-    /// A list of all public keys known to the validator service.
-    pub pubkeys: Vec<PublicKey>,
+    /// A list of all signer objects known to the validator service.
+    // TODO: Generalise the signers, so that they're not just keypairs
+    pub signers: Arc<Vec<Keypair>>,
```
Comment (AgeManning) [inline validator_client/src/duties/mod.rs]: Does this need to be an Arc?

Comment (spble) [reply validator_client/src/duties/mod.rs]: I'm not sure, I made it an Arc because you did in `56d33d2e2`.

**Comment by AgeManning** [comment]
Yep cool. I think if we build our own little block and attestation producers and leave `/eth2` alone, it will be more compatible. We can look to merging them into a single implementation a little later on. 
Unless you can get everything work with the modifications in `/eth2`. 

This looks good tho. 

I'm going to continue on a path for the block producing.

**Comment by paulhauner** [comment]
> Yep cool. I think if we build our own little block and attestation producers and leave `/eth2` alone, it will be more compatible. We can look to merging them into a single implementation a little later on.
> Unless you can get everything work with the modifications in `/eth2`.

This sounds good to me. `test_harness` will likely need a bit of a refactor after all this, so that'll fit in nicely.

File: validator_client/src/duties/mod.rs (hunk 3 above)
Comment (AgeManning) [reply validator_client/src/duties/mod.rs]: Yep, I think we'll borrow and read them in the producer and attestor threads. Should be Arc :)

**Comment by AgeManning** [comment]
I've been refactoring the validator client (again) sorry all... 
I'm going to merge this into my refactoring, and these changes will be reflected in the new PR. 

I will include all the attestation changes and leave the eth2 and testharness un-touched and we can re-address once validator client is complete, so we can have a clearer view of how to modify the block_producer and attester services.

**Pull request closed without merging**